### DISCUSSION TITLE
Language selection not honored

### DISCUSSION THREAD

**Discussion by juergen-de** [OrchardCMS/OrchardCore#8268] [Q&A]
I have content in several language. Switching language for a page works fine, but if I visit another page through a link or go to the homepage by using the domain name, I always get content in the default language. Neither the cookie nor the browser setting is honored.
In the General Settings, I selected to set the language cookie, which looks fine to me if I check in the browser.
Am I missing anything ?
I would expect that as long as content is available in a selected language, that language should remain for the session or until switched.

**Comment by juergen-de** [comment]
Sorry I try to get attention with that again. Anybody who creates multilingual sites who can at least say if I understand something wrong and its working as designed or if I do anything wrong?

**Comment by Skrypt** [comment]
It uses by default the AcceptLanguageHeaderRequestCultureProvider which will set your homepage in the prefered culture set in your browser. If you create a culture picker to set your prefered culture based on a cookie, then the cookie culture will be taken in priority. So, here, you need to create a culture picker on frontend of your website that uses a cookie or simply redirect to ?culture=culturecode when your user comes from a page that is in a specific culture.

There are multiple ways to do this. Using a cookie is not my prefered one since when you add a page in your browser favorites you intent to get back this page in the culture you we're seeing it. If someone cleans his cookies then this page won't be necessarly displayed in the proper culture the next time he visits it. So, using a url which contains a culture code is always more appropriate. So, your frontend culture picker should redirect your users to either a url that has ?culture=culturecode or ?uiculture=culturecode or domain.com/culturecode/something.

**Reply by juergen-de** [reply]
Thank you for the detailed explanation. I understand your thoughts behind using an URL containing the culture code, but prefer the cookie version.
I have a culture picker, I chose to use cookies. Redirection works fine when I switch cultures, but the culture is not regarded on next call, but again, perhaps I understand something wrong.
If I have domain.com/contact for EN, /kontakt for DE and /contacto for ES, what is supposed to happen if I call /contacto but have both browser setting on english and cookie for english language ?

**Reply by Skrypt** [reply]
Ok maybe something needs to be documented here or we need to investigate how it works and how it should.
Though, if you go to /contacto it should display the content in "ES" ... even if you have set your cookie to "EN" for example.
It's not going to automatically reaffect the cookie with a user prefered culture just because you browse in a page in a different culture.
